site/zenodo_rdm/moderation/handlers.py

```diff
@@ -75,106 +58,136 @@ def rules(self):
+                    "User is exempt from moderation", extra={"user_id": user.id}
+                )
+                return
+
             score = 0
             for rule in self.rules:
                 score += rule(identity, draft=draft, record=record)
 
-            # TODO: Move to caller handler method (i.e. publish, create, update, etc.)
-            user_id = None
-            if identity == system_identity:
-                # Use the owner of the record as the user for moderation actions
-                if isinstance(record, records_service.record_cls):
-                    user_id = record.parent.access.owned_by.owner_id
-            else:
-                user_id = identity.id
-            if user_id is None:
-                current_app.logger.error(
-                    "No user found for moderation action", stack_info=True
-                )
-                return
-
-            user = UserAggregate.get_record(user_id)
             action_ctx = {
-                "user_id": user_id,
+                "user_id": user.id,
                 "record_pid": record.pid.pid_value,
                 "score": score,
             }
-            apply_actions = current_app.config.get("MODERATION_APPLY_ACTIONS", False)
+            current_app.logger.info(
+                f"Moderation score for {user.id} calculated: {score}", extra=action_ctx
+            )
             if score > current_scores.spam_threshold:
                 action_ctx["action"] = "block"
-                if apply_actions:
-                    uow.register(
-                        ExceptionOp(
-                            RecordCommitOp(
-                                user,
-                                indexer=current_users_service.indexer,
-                                index_refresh=True,
-                            ),
-                            action_func=lambda *_: user.block(),
-                        )
-                    )
-                    uow.register(
-                        ExceptionOp(
-                            TaskOp(
-                                execute_moderation_actions,
-                                user_id=identity.id,
-                                action="block",
-                            )
-                        )
-                    )
-
-                    raise UserBlockedException()
+                if self.should_apply_actions:
+                    # If user is verified, we need to (re)open the moderation
+                    # request to be reviewed by admins manually again
+                    if user.verified:
+                        action_ctx["action"] = "moderate"
+                        self._moderate(user, uow, action_ctx)
+                    else:
+                        self._block(user, uow, action_ctx)
                 else:
                     current_app.logger.error(
                         "Block moderation action triggered",
                         extra=action_ctx,
                     )
```

Comment:
this might be good to have when we are applying the action as well? in some other way maybe

Reply:
It's logged on the user moderation request as a comment for now (see the tasks file). But yeah, I would prefer later for it to be logged as structured application logging to OpenSearch